app/forms/umrdr/forms/work_form.rb

```diff
@@ -31,5 +29,14 @@ def rights
         @model.rights.first
     end
 
+    # You like danger?! Because you better pass in a hash with the correct keys. 
+    # Can be alleviated when form coverage attribute is:
+    # `coverages: [{begin: {:year, :month, :day}, end: {:year, :month, :day}}, ... ]`
+    def merge_date_coverage_attributes!(hsh)
+      hsh ||= {}
+      hsh.stringify_keys!
+      @attributes.merge! hsh
+    end
```

Comment:
Does this method sometimes get passed `nil` or `false`?

If yes, at first glance the only way to make it a little cleaner would probably be something like:

```ruby
  hsh ||= {}
 @attributes.merge!(hsh.stringify_keys)
```

I am not using the `!` version here because who knows where that hash is coming from and what others are doing with it. Unless you encounter actual performance problems it is best to avoid the mutating versions of methods.